### PR TITLE
Disable materialized view on greenplum.

### PR DESCRIPTION
**Pull request by paul-guo-**
This feature will not be supported in the oncoming gpdb major release.
Explicitly disable it for now.

### CODE REVIEW

**Comment by paul-guo-** [comment]
We should document this.

**Comment by dyozie** [comment]
Looking at the docs, it doesn't seem like we've ever described  away to materialize views in Greenplum:  "CREATE VIEW defines a view of a query. The view is not physically materialized. Instead, the query is run every time the view is referenced in a query."

+ @mkiyama in case I'm missing something here.

**Comment by mkiyama** [comment]
@dyozie 
It looks like a postgres 9.3 feature.
I think the only thing we need to do is list it as not supported.

File: src/test/regress/serial_schedule [hunk 1]
```diff
@@ -98,7 +98,8 @@ ignore: prepared_xacts
 test: privileges
 test: security_label
 test: collate
-test: matview
+# FIXME: matview is not supported on greenplum. Enable the test when the feature is ready.
```
[the remark this answers is not in the document]
Comment (paul-guo-) [reply src/test/regress/serial_schedule]: Good idea. Thanks.

File: src/backend/parser/analyze.c [hunk 2]
```diff
@@ -3236,6 +3236,10 @@ transformCreateTableAsStmt(ParseState *pstate, CreateTableAsStmt *stmt)
 	/* additional work needed for CREATE MATERIALIZED VIEW */
 	if (stmt->relkind == OBJECT_MATVIEW)
 	{
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("materialized view is not supported on greenplum")));
```
[the remark this answers is not in the document]
Comment (paul-guo-) [reply src/backend/parser/analyze.c]: Just want users who do not know the context explicitly know that this is a missing feature on GP, but not PG.

**Comment by ashwinstar** [comment]
On Tue, Jan 8, 2019 at 6:07 PM Paul Guo <notifications@github.com> wrote:

> *@paul-guo-* commented on this pull request.
> ------------------------------
>
> In src/backend/parser/analyze.c
> <https://github.com/greenplum-db/gpdb/pull/6637#discussion_r246239554>:
>
> > @@ -3236,6 +3236,10 @@ transformCreateTableAsStmt(ParseState *pstate, CreateTableAsStmt *stmt)
>  	/* additional work needed for CREATE MATERIALIZED VIEW */
>  	if (stmt->relkind == OBJECT_MATVIEW)
>  	{
> +		ereport(ERROR,
> +				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
> +				 errmsg("materialized view is not supported on greenplum")));
>
> Just want users who do not know the context explicitly know that this is a
> missing feature on GP, but not PG.
>
Well ideally if its missing in PG, even the error will not be produced as
code for the same will be absent in PG to even throw the error (like PG
doesn't give error for appendonly tables not supported :-)) But yes no
problems keeping it either being explicit is better as well.

**Comment by Eulerizeit** [comment]
>Well ideally if its missing in PG, even the error will not be produced as code for the same will be absent in PG to even throw the error (like PG doesn't give error for appendonly tables not supported :-)) But yes no problems keeping it either being explicit is better as well.

As people look and see that we are at Postgres version 9.4.x there will/might reasonably be an expectation that all features for that version are working. It is a little awkward either way.

**Pull request merged**